### PR TITLE
Add a error message when creating an empty Constant

### PR DESCRIPTION
**Pull request by siddhant-0707**
### Details:
 - Added an error message when creating an empty constant to the constant wrapper

### Tickets:
 - #19617

### CODE REVIEW

**Review by p-wysocki — APPROVED** [review]
Thank you for your contribution! LGTM. @siddhant-0707 have you tested it locally to see if the ValueError is raised?

There are currently some issues with CI, please do not worry that some jobs are failing.

**Comment by siddhant-0707** [comment]
@p-wysocki Yes, ValueError raised correctly. I tested it on Python 3.10.12

**Comment by akuporos** [comment]
Can you please also extend the test [here](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_graph/test_ops.py#L122), that error raised?

**Review by akuporos — COMMENTED** [review]
Can you please also extend the test [here](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/python/tests/test_graph/test_ops.py#L122), that error raised?

File: src/bindings/python/tests/test_graph/test_ops.py [hunk 1]
```diff
@@ -117,9 +117,13 @@ def test_broadcast():
...
     Constant(Type.f32, Shape([3, 3]), list(range(9))),
     ov.constant(np.arange(9).reshape(3, 3), Type.f32),
-    ov.constant(np.arange(9).reshape(3, 3), np.float32)
+    ov.constant(np.arange(9).reshape(3, 3), np.float32),
+    None
 ])
 def test_constant(node):
+    if node is None:
+        with pytest.raises(ValueError):
+            ov.constant(None)
```
Comment (akuporos) [inline src/bindings/python/tests/test_graph/test_ops.py:126]:
```suggestion
    ov.constant(np.arange(9).reshape(3, 3), np.float32)
])
def test_constant(node):
    with pytest.raises(ValueError) as e:
        ov.constant(None)
    assert 'Cannot create an empty Constant. Please provide valid data.' in str(e.value)
```

File: src/bindings/python/tests/test_graph/test_ops.py [hunk 2]
```diff
@@ -117,9 +117,13 @@ def test_broadcast():
 @pytest.mark.parametrize("node", [
     Constant(Type.f32, Shape([3, 3]), list(range(9))),
     ov.constant(np.arange(9).reshape(3, 3), Type.f32),
-    ov.constant(np.arange(9).reshape(3, 3), np.float32)
+    ov.constant(np.arange(9).reshape(3, 3), np.float32),
+    None
```
Comment (akuporos) [inline src/bindings/python/tests/test_graph/test_ops.py:121]: I suggest to rewrite a little bit this test using [this feature](https://docs.pytest.org/en/7.1.x/example/parametrize.html#parametrizing-conditional-raising).

```suggestion
    ov.constant(None)
```

File: src/bindings/python/tests/test_graph/test_ops.py [hunk 3]
```diff
@@ -117,9 +117,13 @@ def test_broadcast():
 @pytest.mark.parametrize("node", [
     Constant(Type.f32, Shape([3, 3]), list(range(9))),
     ov.constant(np.arange(9).reshape(3, 3), Type.f32),
```
Comment (akuporos) [inline src/bindings/python/tests/test_graph/test_ops.py:119]:
```suggestion
@pytest.mark.parametrize(("const", "args", "expectation"), [
    (Constant, (Type.f32, Shape([3, 3]), list(range(9))), does_not_raise()),
    (ov.constant, (np.arange(9).reshape(3, 3), Type.f32), does_not_raise()),
```

File: src/bindings/python/tests/test_graph/test_ops.py [hunk 4]
```diff
@@ -117,9 +117,13 @@ def test_broadcast():
...
+    ov.constant(np.arange(9).reshape(3, 3), np.float32),
+    ov.constant(None)
 ])
 def test_constant(node):
+    if node is None:
+        with pytest.raises(ValueError):
+            ov.constant(None)
     assert node.get_type_name() == "Constant"
     assert node.get_output_size() == 1
     assert list(node.get_output_shape(0)) == [3, 3]
```
Comment (akuporos) [inline src/bindings/python/tests/test_graph/test_ops.py:129]:
```suggestion
    (ov.constant, (np.arange(9).reshape(3, 3), np.float32), does_not_raise()),
    (ov.constant, [None], pytest.raises(ValueError)),
])
def test_constant(const, args, expectation):
    with expectation:
        node = const(*args)
        assert node.get_type_name() == "Constant"
        assert node.get_output_size() == 1
        assert list(node.get_output_shape(0)) == [3, 3]
```

Comment (akuporos) [reply src/bindings/python/tests/test_graph/test_ops.py:129]: please, also add indentation to the line with `assert node.get_output_element_type(0) == Type.f32`, this assert need to be under `with expectation`.

and please, add `from contextlib import nullcontext as does_not_raise` in the begging of this file

**Review by akuporos — CHANGES_REQUESTED** [review]

**Review by akuporos — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]

**Comment by siddhant-0707** [comment]
Hey, was wondering if I need to do something else also to merge this PR?

**Comment by akuporos** [comment]
> Hey, was wondering if I need to do something else also to merge this PR?

Actually no, just wait for the last job will pass

**Comment by akuporos** [comment]
build_jenkins

**Pull request merged**